tests/functional/test_install_config.py

```diff
@@ -40,6 +40,19 @@ def test_command_line_options_override_env_vars(script, virtualenv):
     assert "Getting page https://download.zope.org/ppix" in result.stdout
 
 
+def test_no_password_in_debug_message(script, virtualenv):
+    """
+    Test that password in the URL is not logged
+    """
+    script.environ['PIP_INDEX_URL'] = 'https://user:my_password@example.com/simple/'
+    result = script.pip('install', '-vvv', 'INITools', expect_error=True)
+    assert (
+        "Getting page https://user:<pwd>@example.com/simple/initools"
+        in result.stdout
+    )
```

Comment:
In addition, you might as well assert that `my_password` is _not_ in the output. (And I would assign `my_password` to the variable so that that portion of the test doesn't pass because of a typo.)

Reply:
Done